engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -327,8 +327,11 @@ public PhysicalPlan(LogicalPlan dag, PlanContext ctx) {
 
       boolean upstreamDeployed = true;
 
-      for (StreamMeta s : n.getInputStreams().values()) {
-        if (s.getSource() != null && !this.logicalToPTOperator.containsKey(s.getSource().getOperatorMeta())) {
+      for (Map.Entry<InputPortMeta, StreamMeta> entry : n.getInputStreams().entrySet()) {
+        InputPortMeta port = entry.getKey();
+        StreamMeta s = entry.getValue();
+        boolean delay = port.getValue(PortContext.IS_CONNECTED_TO_DELAY_OPERATOR);
```

Comment:
I don't see `port` getting used anywhere else. Can we do `entry.getKey().getValue()`?